nautobot/docs/user-guide/administration/guides/request-profiling.md

```diff
@@ -0,0 +1,85 @@
+# Request Profiling
+
+Nautobot offers advanced request profiling through [django-silk](https://github.com/jazzband/django-silk). This allows administrators to collect debug information about user activities, which can be used to troubleshoot issues with the system.
+
+## User Setting
+
+Request profiling may be enabled by individual users in their profile within the web interface. This can be found under the "Advanced Settings" section.
+
+![user advanced settings](../../../media/user-guide/administration/guides/request-profiling/advanced-settings.png)
+
+Once a user enables request profiling, all subsequent HTTP requests made by that specific user to the system will be logged by django-silk. This setting will persist until the user either logs out or disables the setting in their profile.
+
+## Silk UI
+
+Nautobot administrators with super-user permissions can access the django-silk UI at the `/silk/` URL.
+
+![silk ui](../../../media/user-guide/administration/guides/request-profiling/silk-ui.png)
+
+From there, administrators can view details of individual requests, including timing, SQL queries, and cProfile artifacts.
+
+## Configuration
+
+The Nautobot configuration comes out of the box with django-silk set up to support the above functionality. Those settings are described below, but it is not the intention of the Nautobot docs to describe all django-silk settings. Django-silk provides several other parameters that knowledgeable users may also be able to use, depending on the use case.
+
+### ALLOW_REQUEST_PROFILING
```

Comment:
This one at least should perhaps also be documented in `optional-settings.md`, perhaps with a cross-reference to this file?